File: akka-http-cors/src/main/scala/ch/megard/akka/http/cors/HttpHeaderRange.scala
package ch.megard.akka.http.cors

import java.util.Locale

import scala.collection.immutable.Seq


abstract class HttpHeaderRange extends ch.megard.akka.http.cors.japi.HttpHeaderRange

object HttpHeaderRange {
  case object `*` extends HttpHeaderRange {
    def matches(header: String) = true
  }

  final case class Default(headers: Seq[String]) extends HttpHeaderRange {
    val lowercaseHeaders = headers.map(_.toLowerCase(Locale.ROOT))
    def matches(header: String): Boolean = lowercaseHeaders contains header.toLowerCase(Locale.ROOT)
  }

  def apply(headers: String*): Default = Default(Seq(headers: _*))
}

File: akka-http-cors/src/main/java/ch/megard/akka/http/cors/japi/HttpHeaderRange.java
package ch.megard.akka.http.cors.japi;

import akka.http.impl.util.Util;
import ch.megard.akka.http.cors.HttpHeaderRange$;


/**
 * @see HttpHeaderRanges for convenience access to often used values.
 */
public abstract class HttpHeaderRange {
    public abstract boolean matches(String header);

    public static HttpHeaderRange create(String... headers) {
        return HttpHeaderRange$.MODULE$.apply(Util.convertArray(headers));
    }
}
